enterprise/api/v1/views/enterprise_group.py

```diff
@@ -176,91 +194,101 @@ def assign_learners(self, request, group_uuid):
             customer = group.enterprise_customer
         except models.EnterpriseGroup.DoesNotExist as exc:
             raise Http404 from exc
-        if requested_emails := request.POST.dict().get('learner_emails'):
-            total_records_processed = 0
-            total_existing_users_processed = 0
-            total_new_users_processed = 0
-            for user_email_batch in utils.batch(requested_emails.rstrip(',').split(',')[: 1000], batch_size=200):
-                user_emails_to_create = []
-                memberships_to_create = []
-                # ecus: enterprise customer users
-                ecus = []
-                # Gather all existing User objects associated with the email batch
-                existing_users = User.objects.filter(email__in=user_email_batch)
-
-                # Build and create a list of EnterpriseCustomerUser objects for the emails of existing Users
-                # Ignore conflicts in case any of the ent customer user objects already exist
-                ecu_by_email = {
-                    user.email: models.EnterpriseCustomerUser(
-                        enterprise_customer=customer, user_id=user.id, active=True
-                    ) for user in existing_users
-                }
-                models.EnterpriseCustomerUser.objects.bulk_create(
-                    ecu_by_email.values(),
-                    ignore_conflicts=True,
-                )
+        param_serializer = serializers.EnterpriseGroupRequestDataSerializer(data=request.data)
+        param_serializer.is_valid(raise_exception=True)
+        act_by_date = param_serializer.data['act_by_date']
+        catalog_uuid = param_serializer.data['catalog_uuid']
+        learner_emails = param_serializer.data['learner_emails']
+        total_records_processed = 0
+        total_existing_users_processed = 0
+        total_new_users_processed = 0
+        for user_email_batch in utils.batch(learner_emails[: 1000], batch_size=200):
+            user_emails_to_create = []
+            memberships_to_create = []
+            # ecus: enterprise customer users
+            ecus = []
+            # Gather all existing User objects associated with the email batch
+            existing_users = User.objects.filter(email__in=user_email_batch)
```

Comment:
This seems like a red flag to me based on all the problems Alex has with filtering and querying the User table and it taking so long. I would look into this PR and see what he's doing with the SQL string https://github.com/openedx/edx-enterprise/pull/2057/files#diff-d950991245f878f58adc157895d6431af4385d522ab994aa25726e9abc2b434dR4274

Reply:
Though he was doing `email__icontains` and not `email__in` which would be faster but maybe it should be something more like the [_get_filtered_ecu_ids](https://github.com/openedx/edx-enterprise/blob/aa09eea59ce9405332208b2e8ba2888038fddea0/enterprise/models.py#L4274)